src/fondant/compiler.py

```diff
@@ -289,6 +290,30 @@ def compile(
         pipeline.validate(run_id=run_id)
         logger.info(f"Compiling {pipeline.name} to {output_path}")
 
+        def set_component_exec_args(
```

Comment:
This was previously defined in the ComponentSpec but in fact should be evaluated at runtime. Optional arguments with defaults should not be included in the arg section of the componentOp otherwise some errors occur 

```python
I1016 10:42:19.355692      27 cache.go:116] Connecting to cache endpoint 10.0.12.128:8887
[2023-10-16 10:42:21,330 | fondant.cli | INFO] Component `LoadFromHubComponent` found in module main
usage: fondant [-h] [--component_spec COMPONENT_SPEC] [--cache CACHE]
               [--input_partition_rows INPUT_PARTITION_ROWS]
               [--cluster_type CLUSTER_TYPE] [--client_kwargs CLIENT_KWARGS]
fondant: error: argument --input_partition_rows: invalid int value: "{{$.inputs.parameters['input_partition_rows']}}"
```